dart/lib/src/protocol/sentry_user.dart

```diff
@@ -25,15 +27,17 @@ import 'package:meta/meta.dart';
 /// ```
 @immutable
 class SentryUser {
-  /// At a minimum you must set an [id] or an [ipAddress].
+  /// You should provide at least one of [id], [email], [ipAddress], [username]
+  /// for Sentry to be able to tell you how many users are affected by one
+  /// issue, for example. Sending a user that has none of these attributes and
+  /// only custom attributes is valid, but not as useful.
   SentryUser({
     this.id,
     this.username,
     this.email,
     this.ipAddress,
     Map<String, dynamic>? extras,
-  })  : assert(id != null || ipAddress != null),
```

Comment:
since at least one should be provided, what's about adding assert for all the fields with the `or` operator?

https://develop.sentry.dev/sdk/event-payloads/user/